### PR TITLE
Log rule changes through API.

### LINKED ISSUE #7789: Implement logging for alert rule changes made through the API

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
This change should make it easier to detect changes to rules via the API.

**Which issue(s) this PR fixes**:
Fixes #7789

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the `CONTRIBUTING.md` guide
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: pkg/ruler/base/api.go [hunk 1]
```diff
@@ -491,21 +493,23 @@ func (a *API) CreateRuleGroup(w http.ResponseWriter, req *http.Request) {
...
-	level.Debug(logger).Log("msg", "attempting to store rulegroup", "userID", userID, "group", rgProto.String())
+	level.Debug(logger).Log("msg", "attempting to store rulegroup", "group", rgProto.String())
 	err = a.store.SetRuleGroup(req.Context(), userID, namespace, rgProto)
 	if err != nil {
 		level.Error(logger).Log("msg", "unable to store rule group", "err", err.Error())
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
+	level.Info(logger).Log("msg", "storeed rulegroup", "group", rgProto.String())
```
Comment (chaudum) [inline pkg/ruler/base/api.go:511]:
```suggestion
	level.Info(logger).Log("msg", "stored rulegroup", "group", rgProto.String())
```

File: pkg/ruler/base/api.go [hunk 2]
```diff
@@ -448,14 +448,16 @@ func (a *API) CreateRuleGroup(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	logger = log.With(logger, "namespace", namespace, "tenant", userID, "userID", userID)
```
Comment (chaudum) [inline pkg/ruler/base/api.go:451]: Is it really necessary to add the new `tenant` label? I assume it is added to be more consistent across logs, and the old `userID` is kept to be backwards compatible?

Comment (dannykopping) [reply pkg/ruler/base/api.go:451]: I believe we normally log the tenant ID under the `user` field

**Review by dannykopping — COMMENTED** [review]
LGTM but agree with @chaudum's nits

File: pkg/ruler/base/api.go (hunk 2 above)
Comment (chaudum) [reply pkg/ruler/base/api.go:451]: Unfortunately, we are pretty inconsistent with that. We have `user`, `org_id`, `userID`, `tenant`.

Comment (dannykopping) [reply pkg/ruler/base/api.go:451]: I _think_ we use `user` the most, but yes we need to make this consistent. We have a task for including the tenant ID in all log lines, and in that task we can make it consistent.

In any case, I don't see the need for both `tenant` and `userID` here.

Comment (jeschkies) [reply pkg/ruler/base/api.go:451]: > Unfortunately, we are pretty inconsistent with that. We have user, org_id, userID, tenant.

That was my thought. I'll remove `tenant`

**Review by dannykopping — APPROVED** [review]
LGTM

**Review by JordanRushing — APPROVED** [review]
LGTM! Thanks 👍

**Pull request merged**